docs/content/docs/modules/overview.md

```diff
@@ -61,3 +61,36 @@ spec:
 
 A module YAML file can contain multiple YAML documents, separated by `---`, each representing a component to be included in the application.
 Each component is defined by a kind typename string and a spec object containing the component's properties.
+
+# Configuration string interpolation
+You can use `${placeholders}` inside `spec` elements. These will be replaced by entries from a configuration map, consisting of: 
+1. System properties
+2. Environment variables
+3. flink-conf.yaml entries with prefix 'statefun.module.global-config.' 
+4. Command line args
```

Comment:
I think it is more common to give env variables precedence over `flink-conf.yaml` values.

Reply:
I agree in principal. That said, because `globalConfiguration` is already an combination of args and flink-conf.yaml entries with the `statefun.module.global-config.` prefix, there's no easy way to put env variables in between them without affecting other parts of the system. 

I made a start in my fork and the number of changes is pretty high for what we gain. Please have a look and let me know if I Should working on the change you mentioned in this comment

https://github.com/FilKarnicki/flink-statefun/commit/02cd6a9553c74d464a0c4aeadf53d6930b52650e